packages/api/tests/lib/writeRecords/test-write-granules.js

```diff
@@ -503,7 +614,402 @@ test.serial('writeGranulesFromMessage() saves the same values to DynamoDB, Postg
+  );
+  originalEsGranule.files.sort(
+    (f1, f2) => sortFilesByBuckets(f1, f2)
+  );
+
+  // Files were written correctly in initial DB writes
+  t.true(originalPayloadFiles.length > 0);
+  t.deepEqual(originalDynamoGranule.files, originalPayloadFiles);
+  t.deepEqual(originalEsGranule.files, originalPayloadFiles);
+  t.deepEqual(originalApiGranule.files, originalPayloadFiles);
+
+  // Update existing granule with a partial granule object
+  const updateGranulePayload = {
+    granuleId,
+    collectionId: granule.collectionId,
+    cmrLink: 'updatedGranuled.com', // Only field we're changing
+    status: granule.status,
+    // files is undefined
+  };
+
+  const {
+    pgGranule,
+    esGranule,
+    dynamoGranule,
+  } = await updateGranule(t, updateGranulePayload, true);
+
+  const apiGranule = await translatePostgresGranuleToApiGranule({
+    granulePgRecord: pgGranule,
+    knexOrTransaction: knex,
+  });
+
+  esGranule.files.sort(
+    (f1, f2) => sortFilesByBuckets(f1, f2)
+  );
+  dynamoGranule.files.sort(
+    (f1, f2) => sortFilesByBuckets(f1, f2)
+  );
+  apiGranule.files.sort(
+    (f1, f2) => sortFilesByBuckets(f1, f2)
+  );
+
+  t.deepEqual(apiGranule.files, originalPayloadFiles);
+  t.deepEqual(dynamoGranule.files, originalPayloadFiles);
+  t.deepEqual(esGranule.files, originalPayloadFiles);
+});
+
+test.serial('writeGranulesFromMessage() given a partial granule updates only provided fields', async (t) => {
+  const {
+    collectionCumulusId,
+    esGranulesClient,
+    granule,
+    granuleModel,
+    granulePgModel,
+    knex,
+    executionCumulusId,
+    providerCumulusId,
+    granuleId,
+  } = t.context;
+
+  // Need a message in 'completed' state to allow files writes
+  const completedCumulusMessage = {
+    cumulus_meta: {
+      workflow_start_time: t.context.workflowStartTime,
+      state_machine: t.context.stateMachineArn,
+      execution_name: t.context.executionName,
+    },
+    meta: {
+      status: 'completed',
+      collection: t.context.collection,
+      provider: t.context.provider,
+    },
+    payload: {
+      granules: [t.context.granule],
+    },
+  };
+
+  await writeGranulesFromMessage({
+    cumulusMessage: completedCumulusMessage,
+    executionCumulusId,
+    providerCumulusId,
+    knex,
+    granuleModel,
+  });
+
+  t.true(await granuleModel.exists({ granuleId }));
+  t.true(await granulePgModel.exists(
+    knex,
+    { granule_id: granuleId, collection_cumulus_id: collectionCumulusId }
+  ));
+  t.true(await esGranulesClient.exists(granuleId));
+
+  const originalpgGranule = await granulePgModel.get(
+    knex,
+    { granule_id: granuleId, collection_cumulus_id: collectionCumulusId }
+  );
+
+  // Update existing granule with a partial granule object
+  const updateGranulePayload = {
+    granuleId,
+    collectionId: granule.collectionId,
+    cmrLink: 'updatedGranuled.com', // Only field we're changing
+    status: granule.status,
+  };
+
+  const {
+    updatedPgGranuleFields,
```

Comment:
Nit: is there a reason this isn't something like apiGranule?

Reply:
`updatedPgGranuleFields` is meant to be combined with `originalpgGranule` and compared to a new `pgGranule` to make sure that what's in Postgres matches the original + our updates. It's a bit weird. `updatedPgGranuleFields` represents a Postgres granule where the only fields are the fields present in the update payload if that makes any sense at all.

The assertion looks like this:
```
// Postgres granule matches expected updatedGranule
  t.deepEqual(
    omit(removeNilProperties(pgGranule), cumulusMessageOmitList),
    omit(
      removeNilProperties({ ...originalpgGranule, ...updatedPgGranuleFields }),
      cumulusMessageOmitList
    )
  );
```
I want to compare the Postgres granule without a translation to another Postgres-formatted record. Our API <> Postgres translation makes that confusing because the payload isn't in the Postgres format.